src/hexkit/providers/mongodb/provider.py

```diff
@@ -396,10 +431,16 @@ def __init__(
             config: MongoDB-specific config parameters.
         """
         self._config = config
+        timeout_ms = (
+            int(config.mongo_timeout * 1000)
+            if config.mongo_timeout is not None
+            else None
+        )
 
         # get a database-specific client:
         self._client: AsyncIOMotorClient = AsyncIOMotorClient(
-            str(self._config.mongo_dsn.get_secret_value())
+            str(self._config.mongo_dsn.get_secret_value()),
+            timeoutms=timeout_ms,
```

Comment:
Shouldn't this be `timeoutMS` - see [docs](https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html)? Why is this working though?

Reply:
They parse the kwargs with a case-insensitive dictionary

Reply:
Probably to be compatible with mongo clients in other languages. I think we should use the documented spelling though.

Reply:
Ok. The main thing is that we make it possible to catch any DAO related errors with a single exception statement.

Reply:
I know it's actually not in the scope of this PR but since we are already working on error handling, we may want to improve this as well.

Reply:
I will move this extra error translation into a subsequent PR.